service/test/ServiceIOGroupTest.cpp

```diff
@@ -181,6 +181,37 @@ TEST_F(ServiceIOGroupTest, read_signal_behavior)
                                "BAD SIGNAL not valid for ServiceIOGroup");
 }
 
+TEST_F(ServiceIOGroupTest, read_signal_exception)
+{
+    // !is_valid_signal(signal_name)
+    GEOPM_EXPECT_THROW_MESSAGE(
+        m_serviceio_group->read_signal("NUM_VACUUM_TUBES", 4, 0),
+        GEOPM_ERROR_INVALID,
+        "ServiceIOGroup::read_signal(): signal name \""
+    );
```

Comment:
```suggestion
    GEOPM_EXPECT_THROW_MESSAGE(m_serviceio_group->read_signal("NUM_VACUUM_TUBES", 4, 0),
                               GEOPM_ERROR_INVALID, "ServiceIOGroup::read_signal(): signal name");
```

Please use this style for spacing and line endings.  Please change all invocations of ```GEOPM_EXPECT_THROW_MESSAGE``` to match this style.  Look at how other tests use ```GEOPM_EXPECT_THROW_MESSAGE``` and let me know if you have questions about the style.

Additionally, I changed the string in the last argument slightly.  I removed '\"'.  It seems like you started to set this up to expect a specific message in the exception text (I'm assuming "NUM_VACUUM_TUBES"), but then stopped.  What was the intention here?